runtime/src/qio/qio.c

```diff
@@ -1525,6 +1525,12 @@ qioerr _qio_channel_makebuffer_unlocked(qio_channel_t* ch)
 
   qbuffer_reposition(&ch->buf, start);
 
+  // Since we are just making the buffer, make sure that the
+  // available position is to the right of the start position.
+  if( ch->av_end < start ) {
+    ch->av_end = start;
```

Comment:
There is a nice picture around line 774 in qio.h. av_end represents the end of the "available" data. When reading, we read a big chunk (ahead of what the user requested - and av_end is the end of that read data. "right mark start" is the position that the user has read to, if it comes up later...